lib/pbench/test/unit/server/test_shell_cli.py

```diff
@@ -280,27 +283,24 @@ def generate_crontab_if_necessary(*args, **kwargs) -> int:
         assert ret_val == 43
 
     @staticmethod
-    @pytest.mark.parametrize("init_indexing_exc", ["section", "option"])
+    @pytest.mark.parametrize(
+        "init_indexing_exc",
+        [NoSectionError("missingsection"), NoOptionError("section", "missingoption")],
+    )
     def test_main_init_indexing_failed(
         monkeypatch, make_logger, mock_get_server_config, init_indexing_exc
     ):
-        def noop(*args, **kwargs):
+        def immediate_success(*args, **kwargs):
             pass
 
         called = [False]
 
         def init_indexing(*args, **kwargs) -> int:
             called[0] = True
-            if init_indexing_exc == "section":
-                exc = NoSectionError("missingsection")
-            elif init_indexing_exc == "option":
-                exc = NoOptionError("section", "missingoption")
-            else:
-                exc = Exception(f"Bad test parameter, {init_indexing_exc}")
-            raise exc
+            raise init_indexing_exc
 
         monkeypatch.setattr(shell.site, "ENABLE_USER_SITE", False)
-        monkeypatch.setattr(shell, "wait_for_uri", noop)
+        monkeypatch.setattr(shell, "wait_for_uri", immediate_success)
         monkeypatch.setattr(shell, "init_indexing", init_indexing)
 
         ret_val = shell.main()
```

Comment:
Shouldn't we be mocking out `init_db()` in this test?  (And, `Database.create_if_missing()`?)

Ditto for the other tests in which execution gets that far....

Reply:
I don't think we need to do that here.  We have already been relying on the use of sqlite for the success path, and we already mock it out for the error path.